--- diff0.py	2017-04-02 20:46:54.000000000 -0700
+++ diff.py	2017-04-02 20:48:44.000000000 -0700
@@ -232,7 +232,7 @@
     # precompiled regex
     re_header = re.compile(br"""
                                 ^diff[ ]--git
-                                    [ ](?P<a_path_fallback>"?a/.+?"?)[ ](?P<b_path_fallback>"?b/.+?"?)\n
+                                    [ ](?P<a_path_fallback>"?b/.+?"?)[ ](?P<b_path_fallback>"?a/.+?"?)\n
                                 (?:^old[ ]mode[ ](?P<old_mode>\d+)\n
                                    ^new[ ]mode[ ](?P<new_mode>\d+)(?:\n|$))?
                                 (?:^similarity[ ]index[ ]\d+%\n
